src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -10528,11 +10623,831 @@ static void verify(CSharpCompilation comp)
+                //         M1(out scoped scoped _);
+                Diagnostic(ErrorCode.ERR_ScopedDiscard, "scoped").WithLocation(21, 16),
+                // (21,23): error CS0246: The type or namespace name 'scoped' could not be found (are you missing a using directive or an assembly reference?)
+                //         M1(out scoped scoped _);
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "scoped").WithArguments("scoped").WithLocation(21, 23)
+                );
+            verify(comp);
+
+            static void verify(CSharpCompilation comp)
+            {
+                var tree = comp.SyntaxTrees[0];
+                var model = comp.GetSemanticModel(tree);
+                var decls = tree.GetRoot().DescendantNodes().OfType<SingleVariableDesignationSyntax>().ToArray();
+                var locals = decls.Select(d => model.GetDeclaredSymbol(d).GetSymbol<LocalSymbol>()).ToArray();
+
+                Assert.Equal(6, locals.Length);
+
+                VerifyLocalSymbol(locals[0], "scoped R r1", RefKind.None, DeclarationScope.ValueScoped);
+                VerifyLocalSymbol(locals[1], "scoped R r2", RefKind.None, DeclarationScope.ValueScoped);
+                VerifyLocalSymbol(locals[2], "scoped R r5", RefKind.None, DeclarationScope.ValueScoped);
+                VerifyLocalSymbol(locals[3], "scoped R r11", RefKind.None, DeclarationScope.ValueScoped);
+                VerifyLocalSymbol(locals[4], "scoped R r21", RefKind.None, DeclarationScope.ValueScoped);
+                VerifyLocalSymbol(locals[5], "scoped R r51", RefKind.None, DeclarationScope.ValueScoped);
+
+                foreach (var decl in decls)
+                {
+                    var type = ((DeclarationExpressionSyntax)decl.Parent).Type;
+
+                    Assert.True(SyntaxFacts.IsInTypeOnlyContext(type));
+                    Assert.True(SyntaxFacts.IsInTypeOnlyContext(type.SkipScoped(out _)));
+                    Assert.True(SyntaxFacts.IsInTypeOnlyContext(type.SkipScoped(out _).SkipRef(out _)));
+
+                    Assert.Null(model.GetSymbolInfo(type).Symbol);
+                    Assert.Null(model.GetTypeInfo(type).Type);
+
+                    type = type.SkipScoped(out _);
+
+                    if (type is RefTypeSyntax refType)
+                    {
+                        Assert.Null(model.GetSymbolInfo(type).Symbol);
+                        Assert.Null(model.GetTypeInfo(type).Type);
+
+                        type = refType.Type;
+                    }
+
+                    Assert.Equal("R", model.GetSymbolInfo(type).Symbol.ToTestDisplayString());
+                    Assert.Equal("R", model.GetTypeInfo(type).Type.ToTestDisplayString());
+                }
+
+                var discard = tree.GetRoot().DescendantNodes().OfType<DiscardDesignationSyntax>().ToArray();
+                Assert.Equal(8, discard.Length);
+
+                for (int i = 0; i < 6; i++)
```

Comment:
I didn't understand why the last 2 discard elements are not checked (since length was asserted to be 8 above).

Reply:
> I didn't understand why the last 2 discard elements are not checked (since length was asserted to be 8 above).

The last two discards refer to an undeclared type, do not align with the rest, and are not really interesting for the purpose of the test.